### ISSUE TITLE
Evaluate auto build on labwc commits touching man pages

### ISSUE THREAD

**Issue by Consolatis** [labwc/labwc.github.io#9]
It might be possible to add some action / GH workflow to the main labwc repo so that this repo is rebuilt on pushes in labwc (e.g. accepting PRs) that modified the man pages.

**Comment by johanmalm** [comment]
Sounds a good plan.

**Comment by Consolatis** [comment]
Ref: https://stackoverflow.com/a/65514259

Seems like it could be done, requires a GH token in the `public_repo` scope.
I'll cook something up with some test repos so no immediate action is required here.

**Comment by Consolatis** [comment]
Rough sketch for a CI build:

```bash
#!/usr/bin/env bash


: ${DEPLOY_DIR:=.}

build_src_pages () {
	for md in src/*.md; do
		html=${md##*/}
		html="${DEPLOY_DIR}/${html/%.md/.html}"
		echo "Converting $md to $html"

		cat src/template_pre.html > "$html"
		pandoc --from=markdown+footnotes "$md" >> "$html"
		cat src/template_post.html >> "$html"
	done
}

build_man_pages () {
	git clone --branch master --depth 1 https://github.com/labwc/labwc labwc

	for scd in labwc/docs/*.scd; do
		html=${scd##*/}
		html="${DEPLOY_DIR}/${html/%.scd/.html}"
		echo "Converting $scd to $html"

		sed 's/sans-serif;/monospace;/' src/template_pre.html > "$html"
		scdoc < "$scd" |  pandoc --from man >> "$html"
		cat src/template_post.html >> "$html"

		./fix-bullet-formatting.py "$html"
	done
}

main () {
	mkdir -p "$DEPLOY_DIR"
	build_src_pages
	build_man_pages
}

main "$@"
```

This looks interesting as well: https://github.com/rossjrw/pr-preview-action
That would allow to have a preview in PRs (in this repo only)

**Comment by johanmalm** [comment]
Looks good. Slightly simpler to read than the build-site script too.

**Comment by Consolatis** [comment]
> CI looks promising.
Will we still be able to build html to view locally before pushing branches?

I'd just add the new build script but keep the old one in place so that should keep working.
The new one basically only adds the git clone, compiles the manual pages via scdoc and splits the template into a `_pre` and `_post` variant. So I guess we could consolidate the two at some point as well.

**Comment by Consolatis** [comment]
Should be fixed by labwc/labwc#800
